tree/tree/src/TLeafElement.cxx

```diff
@@ -52,6 +53,47 @@ TLeafElement::TLeafElement(TBranch *parent, const char *name, Int_t id, Int_t ty
       if ((bareType >= TVirtualStreamerInfo::kUChar && bareType <= TVirtualStreamerInfo::kULong)
           || bareType == TVirtualStreamerInfo::kULong64)
       SetUnsigned();
+
+      auto bareTypeCopy = static_cast<EDataType>(bareType);
```

Comment:
Why is fLenType not already set correctly?

Reply:
Not clear to me.  It is a member of `TLeaf`, but is only used / set for the primitive sub-classes (`TLeafI`, `TLeafD`, etc) and has gone unused by `TLeafElement`.

Luckily, `TLeaf::GenerateOffsetArrayBase` is correct for `TLeafElement` as long as `fLenType` is set appropriately.

`fLenType` is unneeded for deserialization because `TLeafElement` invokes the streamer, I think:

https://github.com/root-project/root/blob/master/tree/tree/inc/TLeafElement.h#L50